From 19fc4c5a72eb9919d720ad66734029d9f8e313b1 Mon Sep 17 00:00:00 2001
From: Aaron Plattner <aplattner@nvidia.com>
Date: Tue, 12 Jun 2012 12:32:19 -0700
Subject: [PATCH 1/4] Add a "GUID" property

This property represents the globally-unique identifier of the connected
display, if one is available.

Signed-off-by: Aaron Plattner <aplattner@nvidia.com>
Reviewed-by: Andy Ritger <aritger@nvidia.com>
---
 randr.h        |  1 +
 randrproto.txt | 12 ++++++++++++
 2 files changed, 13 insertions(+)

diff --git a/randr.h b/randr.h
index 33d22fb..3c6721b 100644
--- a/randr.h
+++ b/randr.h
@@ -180,6 +180,7 @@ typedef unsigned long	XRandrModeFlags;
 #define RR_PROPERTY_CLONE_LIST		"CloneList"
 #define RR_PROPERTY_BORDER		"Border"
 #define RR_PROPERTY_BORDER_DIMENSIONS	"BorderDimensions"
+#define RR_PROPERTY_GUID		"GUID"
 
 /* roles this device can carry out */
 #define RR_Capability_None 0
diff --git a/randrproto.txt b/randrproto.txt
index 2bf065f..f82407c 100644
--- a/randrproto.txt
+++ b/randrproto.txt
@@ -1957,6 +1957,17 @@ doesn't handle a mandatory property correctly.
 	  2 = left/right and top/bottom borders can be specified independently
 	  4 = all four borders can be specified independently
 
+    "GUID"			aka RR_PROPERTY_GUID
+	Type:			INTEGER
+	Format:			8
+	Num items:		16
+	Flags:			Immutable
+	Range/List:		-
+
+	Some display devices, such as DisplayPort 1.2 devices, have globally
+	unique identifiers.  When such an identifier is available, this property
+	contains its raw bytes.
+
 
 9.2 Properties introduced with version 1.2 of the RandR extension
 
@@ -1990,6 +2001,7 @@ Property			Immutable	Mandatory since
 ────────			─────────	───────────────
 Border				yes		not mandatory
 BorderDimensions		yes: static	not mandatory
+GUID				yes		not mandatory
 
 			      ❧❧❧❧❧❧❧❧❧❧❧
 
-- 
1.9.0

